pkg/core/player/player.go

```diff
@@ -185,10 +215,23 @@ func (h *Handler) ApplyHitlag(char int, factor, dur float64) {
 	if char != h.active {
 		return
 	}
+
 	h.chars[char].ApplyHitlag(factor, dur)
+
 	//also extend infusion
 	//TODO: this is a really awkward place to apply this
 	h.ExtendInfusion(char, factor, dur)
+
+	// extend the dash cd by the hitlag extension amount
+	if h.DashCDExpirationFrame > *h.F {
+		ext := int(math.Ceil(dur * (1 - factor)))
+		h.DashCDExpirationFrame += ext
+		h.Log.NewEvent("dash cd hitlag extended", glog.LogHitlagEvent, char).
+			Write("extension", ext).
+			Write("expiry", h.DashCDExpirationFrame-*h.F).
+			Write("expiry_frame", h.DashCDExpirationFrame).
+			Write("lockout", h.DashLockout)
```

Comment:
nit: Similarly to the comment in stam.go, the log message should probably be different depending on whether it's the 0.8s or 1.5s timer.